File: 06_BUFFER.py
import arcpy

# Set the workspace (path to the directory with your CSV and output shapefiles)
arcpy.env.workspace = r"C:/Users/alvarece/Desktop/MODELAMIENTO Y PROGRAMACION GIS/PRACTICAS/DATOS"

# Input CSV and output shapefiles
csv_file = "POINTS.csv"  # CSV file with the points
output_shapefile = "points.shp"
buffer_shapefile = "buffer_A_C.shp"  # Output buffer shapefile

# Define the fields in the CSV that contain the X (Longitude) and Y (Latitude) coordinates
arcpy.management.XYTableToPoint(csv_file, output_shapefile, "X", "Y", "", 32717)

expression = "Type IN ('A', 'C')" # Create a query to select Type = "A" or "C"
arcpy.management.MakeFeatureLayer(output_shapefile, "points_lyr")  # Make a feature layer from the points shapefile
arcpy.management.SelectLayerByAttribute("points_lyr", "NEW_SELECTION", expression)
arcpy.analysis.Buffer("points_lyr", buffer_shapefile, "500 METERS") # Create a buffer around the selected points (500 meters)

File: 06_BUFFER.R
library(sf, dplyr, ggplot2)
setwd("C:/Users/alvarece/Desktop/MODELAMIENTO Y PROGRAMACION GIS/PRACTICAS/DATOS")
data <- read.csv("POINTS.csv")
data$X <- as.numeric(data$X)
data$Y <- as.numeric(data$Y)
points <- st_as_sf(data, coords = c("X", "Y"), crs = 32717)
points_A <- points %>% filter(Type %in% c("A", "C")) # Filter by the 'Type' attribute where Type is "A"
buffer_A <- st_buffer(points_A, dist = 2000)  # Set the buffer distance as needed
st_write(buffer_A, "buffer_A.shp", delete_layer = TRUE)

ggplot() +
  geom_sf(data = points, color = "red", size = 3) +  # Plot the points
  geom_sf(data = buffer_A, fill = "lightblue", alpha = 0.5) +  # Plot the buffer with transparency
  theme_minimal() +
  labs(title = "Points and Buffer Shapefiles",
       x = "Longitude", 
       y = "Latitude") 